easybuild/tools/toolchain/compiler.py

```diff
@@ -146,8 +146,11 @@ def set_options(self, options):
 
     def set_variables(self):
         """Set the variables"""
+        if self.arch is None:
+            self.arch = systemtools.get_cpu_architecture()
```

Comment:
Not seeing any other use of `.arch` elsewhere is good, but we don't control *all* easyblocks out there, and people have scripts around that hook into the EasyBuild framework, who knows...

That shouldn't hold us back from making this change though, since it makes a whole lot of sense, and if anyone relies on `.arch` somewhere, they probably shouldn't be anyway (they should call the appropriate function from `systemtools` instead).

I see no reason either for lazy init of `self.arch`, so feel free to change that.